enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/IdentityVerificationLimitService.java

```diff
@@ -124,45 +123,51 @@ public void checkIdentityVerificationLimit(OwnerId ownerId) throws RemoteCommuni
     }
 
     /**
-     * Check the limit for maximum number of document uploads.
+     * Reset the given identity verification if the limit for maximum number of document uploads reached.
+     * In that case, thrown of {@link IdentityVerificationLimitException} is expected.
+     *
      * @param ownerId Owner identifier.
+     * @param identityVerification Identity verification.
      * @throws IdentityVerificationLimitException Thrown when document upload limit is reached.
      * @throws RemoteCommunicationException Thrown when communication with PowerAuth server fails.
      * @throws IdentityVerificationException Thrown when identity verification is invalid.
      * @throws OnboardingProcessLimitException Thrown when maximum failed attempts for identity verification have been reached.
      * @throws OnboardingProcessException Thrown when onboarding process is invalid.
      */
-    public void checkDocumentUploadLimit(OwnerId ownerId, IdentityVerificationEntity identityVerification) throws IdentityVerificationLimitException, RemoteCommunicationException, IdentityVerificationException, OnboardingProcessLimitException, OnboardingProcessException {
+    public void checkDocumentUploadLimit(OwnerId ownerId, IdentityVerificationEntity identityVerification)
+            throws IdentityVerificationLimitException, RemoteCommunicationException, IdentityVerificationException, OnboardingProcessLimitException, OnboardingProcessException {
         final List<DocumentVerificationEntity> documentVerificationsFailed = documentVerificationRepository.findAllDocumentVerifications(identityVerification, DocumentStatus.ALL_FAILED);
         if (documentVerificationsFailed.size() > config.getDocumentUploadMaxFailedAttempts()) {
-            final IdentityVerificationPhase phase = identityVerification.getPhase();
-
-            identityVerification.setStatus(IdentityVerificationStatus.FAILED);
-            identityVerification.setErrorDetail(IdentityVerificationEntity.ERROR_MAX_FAILED_ATTEMPTS_DOCUMENT_UPLOAD);
-            identityVerification.setErrorOrigin(ErrorOrigin.PROCESS_LIMIT_CHECK);
-            identityVerification.setTimestampLastUpdated(ownerId.getTimestamp());
-            identityVerification.setTimestampFailed(ownerId.getTimestamp());
-            identityVerificationRepository.save(identityVerification);
-            logger.info("Switched to {}/FAILED; {}", phase, ownerId);
-            auditService.audit(identityVerification, "Switched to {}/FAILED; user ID: {}", phase, ownerId.getUserId());
-            resetIdentityVerification(ownerId);
+            resetIdentityVerification(ownerId, ErrorOrigin.PROCESS_LIMIT_CHECK, IdentityVerificationEntity.ERROR_MAX_FAILED_ATTEMPTS_DOCUMENT_UPLOAD);
```

Comment:
Not sure whether `ErrorOrigin.PROCESS_LIMIT_CHECK` is correct for `documentUploadMaxFailedAttempts`, but it was already there.

Reply:
Yes, it is fine, all limit checks are under this one.